WIP: Update attribute arguments tables

keerlu opened the pull request:
## Describe this PR

This is an initial version of the schema reference information to go with #2514. Would appreciate some comments:

- there were some changes to the signatures in 3.0.0 that didn't make it into the table of arguments, so I've attempted to add them but it may be incorrect or missing descriptions
- the new changes are version- and database-specific so it looks a little messy, there might be better ways of doing it



## Changes

<!-- What changes have you made? Keep it brief!

- Refactored example to include new database field
- ... -->

## What issue does this fix?

<!-- Link the issue this is solving, if applicable, using the issue number. This can be found to the right of the issue title, or in the url.

  -->

## Any other relevant information

<!-- Add any other information you deem to be relevant to the PR -->

keerlu commented:
The build is failing and marking the link /docs/concepts/components/prisma-schema/index-configuration as broken, not sure why as [this page](https://www.prisma.io/docs/concepts/components/prisma-schema/index-configuration) exists? Also the links are working for me when I run locally

janpio commented:
Yep, quick check shows that the link _should_ be correct @keerlu. You are using the correct format, to a link/file that exists in this branch. Seems the link checker somehow does not understand that. @nilubava will have to double check that unfortunately. I am not aware of a previous time where this was a problem.

keerlu commented:
@janpio thanks - I've updated to relative links, seems happier now.

janpio commented:
That should _not_ be necessary and generally not how most of the links are done in the docs (as it makes moving pages into other places super hard and painful, also writing links is much more effort this way) - but if this solves the problem for your right now, 🤷 (Maybe open another PR that reproduces this problem, so @nilubava can take a look and fix the script checking the links!)

janpio commented:
(Looking forward to the content of the actual PR here when you are ready - super useful)

keerlu commented:
Thanks @janpio, I thought it was weird that I needed to do the links like that.

I think the PR is now ready to review - I had it in draft because of the links problem and some small bits to fix. The main missing thing remaining is some of the descriptions in the attributes tables for v3.0.0 features.

janpio commented on content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:
```diff
@@ -1040,20 +1040,35 @@ Defines a single-field ID on the model.
 #### Arguments
 
-N/A
+| Name  | Required | Type     | Description |
+| ----- | -------- | -------- | ----------- |
+| `map` | **No**   | `String` |             |
```
Good catch, seems this was missing!

janpio commented on content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:
````diff
@@ -1040,20 +1040,35 @@ Defines a single-field ID on the model.
 > @id
 > ```
 
+> **Note**: In version 3.5.0 and later, the signature for the `"extendedIndexes"` Preview feature is:
````
Could be:
```suggestion
> **Note**: In version 3.5.0 and later, with the `"extendedIndexes"` Preview feature enabled, the signature is:
```

janpio commented on content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:
```diff
@@ -1311,6 +1326,8 @@ Defines a multi-field ID (composite ID) on the model.
 | Name     | Required | Type               | Description                                                      |
 | -------- | -------- | ------------------ | ---------------------------------------------------------------- |
 | `fields` | **Yes**  | `FieldReference[]` | A list of field names - for example, `["firstname", "lastname"]` |
+| `fields` | **No**   | `String`           |                                                                  |
+| `fields` | **No**   | `String`           |                                                                  |
```
This seems obsolete/leftover?

keerlu replied:
Thanks, copy paste error where I missed updating the argument names

janpio commented:
In total I am surprised the typos and missing old information was not fixed before, as I was under the impression it was. Good you did that.

Maybe ask for additional full reviews of the PR in #team-migrations 👍

do4gr commented on content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:
```diff
@@ -1040,20 +1040,35 @@ Defines a single-field ID on the model.
+| Name     | Required | Type     | Description                                                                                                                 |
+| -------- | -------- | -------- | --------------------------------------------------------------------------------------------------------------------------- |
+| `length` | **No**   | `number` | **MySQL only.** Allows you to specify a maximum length for the subpart of the value to be indexed.                          |
+| `sort`   | **No**   | `String` | **SQL Server only.** Allows you to specify in what order the entries of the index or constraint are stored in the database. |
```
Maybe add that `Desc` and `Asc` are the only supported options.

do4gr commented on content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:
```diff
@@ -1311,6 +1326,8 @@ Defines a multi-field ID (composite ID) on the model.
 | Name     | Required | Type               | Description                                                      |
 | -------- | -------- | ------------------ | ---------------------------------------------------------------- |
 | `fields` | **Yes**  | `FieldReference[]` | A list of field names - for example, `["firstname", "lastname"]` |
+| `name`   | **No**   | `String`           |                                                                  |
```
These could use descriptions: 
name: The name the Client Api will expose for the id argument covering all fields i.e. `fullName: { firstName: Paul, lastName: Panther}`
map: The name of the underlying Primary Key constraint in the database.

do4gr commented on content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:
```diff
@@ -1319,18 +1336,31 @@ The name of the `fields` argument on the `@@id` attribute can be omitted:
+| Name     | Required | Type     | Description                                                                                                                 |
+| -------- | -------- | -------- | --------------------------------------------------------------------------------------------------------------------------- |
+| `length` | **No**   | `number` | **MySQL only.** Allows you to specify a maximum length for the subpart of the value to be indexed.                          |
+| `sort`   | **No**   | `String` | **SQL Server only.** Allows you to specify in what order the entries of the index or constraint are stored in the database. |
```
Same as above, could list the Asc, Desc options.

do4gr commented on content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:
````diff
@@ -1319,18 +1336,31 @@ The name of the `fields` argument on the `@@id` attribute can be omitted:
+> **Note**: In version 3.5.0 and later, the signature for the `"extendedIndexes"` Preview feature is:
+>
+> ```prisma no-lines
+> @id(_ fields: FieldReference[], name: String?, map: String?, length: number?, sort: String?)
````
The signature is a bit different on compounds since length and sort are defined on the field level: 
`@@id(_ fields: FieldReference[], name: String?, map: String?)`
With field reverence now being 
`filename(length: Number?, sort: String?)` If no option is given, no empty braces are rendered, just the name.
Example: `@@id([firstName, middleName(length: 1), lastName(sort:Desc)], name: "fullName", map: "PkConstraint")`

keerlu replied:
Thanks, would this mean that the signature of `@@id` is unchanged for the preview feature, but the signature of `FieldReference[]` has? So should I update that somewhere else?

Also I don't quite follow the 'If no option is given' section, where would the option be given?

do4gr replied:
Yes, I think you're correct, the @@id signature stays the same. 
No option given refers to length or sort, so in my example firstName has no additional options specified and stays as `firstName` instead of `firstName()`.

keerlu replied:
Thanks for clearing that up. I've left the signature as unchanged.

do4gr commented on content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:
````diff
@@ -2013,18 +2060,31 @@ The name of the `fields` argument on the `@@unique` attribute can be omitted:
 
+> **Note**: In version 3.5.0 and later, the signature for the `"extendedIndexes"` Preview feature is:
+>
+> ```prisma no-lines
````
Same as for @@id

do4gr replied:
Also this contains a copypasted @id I think.

keerlu replied:
Oops, thanks for catching the copypastes

do4gr commented on content/400-reference/200-api-reference/100-prisma-schema-reference.mdx:
````diff
@@ -2248,18 +2308,31 @@ The _name_ of the `fields` argument on the `@@index` attribute can be omitted:
+> **Note**: In version 3.5.0 and later, the signature for the `"extendedIndexes"` Preview feature is:
+>
+> ```prisma no-lines
+> @id(_ fields: FieldReference[], map: String?, length: number?, sort: String?)
````
Same as above, this is still @id instead of @@index and the length and sort need to go on the field level

do4gr reviewed (changes requested):
Thanks for starting this, I left some comments with regards to the signatures. Please ask if anything is still unclear.

keerlu commented:
I've made a horrible Git merge mess of this PR, sorry. I've opened a new PR #2613 to replace it - hope this is ok, if so I'll close this one. I should have addressed all the comments now.

janpio commented:
https://github.com/prisma/docs/pull/2613 was merged, so I think we can indeed close this one.
